test/test_nestedtensor.py

```diff
@@ -512,6 +512,12 @@ def test_nested_tensor_indexing(self, device, dtype):
         answer = torch.tensor(200.0, device=device, dtype=dtype).expand(4)
         self.assertEqual(nt[1, 1, :], answer)
 
+        # Test that indexing works when requires_grad_(True)
+        # previously this was failing because the backward kernel for select.int uses .sizes()
+        nt = torch.nested_tensor([x0, x1]).requires_grad_(True)
```

Comment:
Could you also add an autograd test  in the bottom class

Reply:
Which really highlights that it would be great to get nested tensors but working with Opinfos... Ill add it to the workstream list

Reply:
Sorry I approved before this comment  and made you cancel I just noticed it

Reply:
No this was a completely valid comment and there was indeed a bug, ty! :)